test_runner/fixtures/neon_fixtures.py

```diff
@@ -78,7 +78,7 @@
 DEFAULT_BRANCH_NAME: str = "main"
 
 BASE_PORT: int = 15000
-WORKER_PORT_NUM: int = 1000
+WORKER_PORT_NUM: int = 384
```

Comment:
Why 384? Won't it be too little for a sequential run?

Maybe instead of predefined `WORKER_PORT_NUM` we can calculate it in a runtime `(32768 - 15000) / numprocesses`? And it will likely require a runtime check that we're not trying to use ports allocated for the next worker.

Reply:
This was me incorrectly thinking that the WORKER_PORT_NUM only needed to accommodate concurrently used ports, not total ports used for all tests on the worker.  Updated to pick this dynamically instead.